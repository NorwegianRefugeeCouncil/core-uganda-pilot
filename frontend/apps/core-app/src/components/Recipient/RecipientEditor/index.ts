export { RecipientEditorComponent as RecipientEditor } from './RecipientEditor.component';
