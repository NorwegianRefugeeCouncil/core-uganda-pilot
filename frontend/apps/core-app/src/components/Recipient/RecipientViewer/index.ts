export { RecipientViewerComponent as RecipientViewer } from './RecipientViewer.component';
