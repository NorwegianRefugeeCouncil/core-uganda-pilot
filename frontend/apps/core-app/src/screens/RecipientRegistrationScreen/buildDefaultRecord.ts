import {
  FieldKind,
  FormDefinition,
  getFieldKind,
  Record,
} from 'core-api-client';

export const buildDefaultRecord = (form: FormDefinition): Record => ({
  id: '', // We are creating records so there shouldn't be an id
  databaseId: form.databaseId,
  formId: form.id,
  ownerId: undefined,
  values: form.fields.map((field) => {
    const fieldType = getFieldKind(field.fieldType);
    switch (fieldType) {
      case FieldKind.Text:
      case FieldKind.MultilineText:
        return {
          fieldId: field.id,
          value: '',
        };
      case FieldKind.Reference:
      case FieldKind.Date:
      case FieldKind.Month:
      case FieldKind.Week:
      case FieldKind.SingleSelect:
        return {
          fieldId: field.id,
          value: null,
        };
      case FieldKind.MultiSelect:
        return {
          fieldId: field.id,
          value: [],
        };
      case FieldKind.Checkbox:
        return {
          fieldId: field.id,
          value: 'false',
        };
      case FieldKind.SubForm:
        return {
          fieldId: field.id,
          value: [],
        };
      default:
        return {
          fieldId: field.id,
          value: '',
        };
    }
  }),
});
