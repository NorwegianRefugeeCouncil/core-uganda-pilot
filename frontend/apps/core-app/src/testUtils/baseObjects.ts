import { FormDefinition, FormType, Record } from 'core-api-client';

export const baseForm: FormDefinition = {
  id: 'form1',
  code: '',
  name: 'form 1',
  databaseId: 'databaseId1',
  folderId: 'folderId1',
  formType: FormType.DefaultFormType,
  fields: [],
};

export const baseTextField = {
  id: 'field1',
  name: 'field 1',
  description: 'description 1',
  code: '',
  required: false,
  key: false,
  fieldType: {
    text: {},
  },
};

export const baseMultilineTextField = {
  id: 'field2',
  name: 'field 2',
  description: 'description 2',
  code: '',
  required: true,
  key: true,
  fieldType: {
    multilineText: {},
  },
};

export const baseRecord1: Record = {
  id: 'record1',
  values: [
    {
      value: 'text',
      fieldId: 'field1',
    },
    {
      value: 'multi line text',
      fieldId: 'field2',
    },
  ],
  formId: 'form1',
  databaseId: 'databaseId1',
  ownerId: undefined,
};
