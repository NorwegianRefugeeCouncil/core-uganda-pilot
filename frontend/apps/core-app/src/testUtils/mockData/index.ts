export { makeRecord } from './Record';
export { makeForm } from './Form';
export { makeField } from './FormField';
