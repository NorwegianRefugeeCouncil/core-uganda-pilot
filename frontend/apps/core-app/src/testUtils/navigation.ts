export const mockNavigationProp = {
  dispatch: jest.fn(),
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getParent: jest.fn(),
  getState: jest.fn(),
  setParams: jest.fn(),
  setOptions: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
};
