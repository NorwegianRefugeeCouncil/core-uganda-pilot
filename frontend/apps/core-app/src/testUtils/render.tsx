import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import {
  render as r,
  RenderAPI,
  RenderOptions,
} from '@testing-library/react-native';
import { render as rWeb } from '@testing-library/react';
import { NativeBaseProvider } from 'native-base';
import { theme } from 'core-design-system';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

export const render = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
): RenderAPI =>
  r(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );

export const renderWeb = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
) =>
  rWeb(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );
