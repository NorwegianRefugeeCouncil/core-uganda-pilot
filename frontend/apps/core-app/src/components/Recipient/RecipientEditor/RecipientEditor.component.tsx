import * as React from 'react';
import { VStack } from 'native-base';
import { FormWithRecord } from 'core-api-client';
import { Accordion } from 'core-design-system';
import { Recipient } from 'core-api-client/src/types/client/Recipient';

import { RecordEditor } from '../../RecordEditor';

type Props = {
  data: FormWithRecord<Recipient>[];
};

export const RecipientEditorComponent: React.FC<Props> = ({ data }) => (
  <VStack space={4}>
    <Accordion header={data[1].form.name} defaultOpen>
      <VStack space={4}>
        <RecordEditor form={data[0].form} hideKeyFields />
        <RecordEditor form={data[1].form} hideKeyFields />
      </VStack>
    </Accordion>
    {data.slice(2).map(({ form }) => (
      <Accordion key={form.id} header={form.name} defaultOpen>
        <RecordEditor form={form} hideKeyFields />
      </Accordion>
    ))}
  </VStack>
);
