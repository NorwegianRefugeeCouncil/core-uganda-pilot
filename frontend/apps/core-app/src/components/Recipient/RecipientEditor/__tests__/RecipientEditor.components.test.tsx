import {
  FormDefinition,
  FormType,
  FormWithRecord,
  Record,
} from 'core-api-client';
import { Recipient } from 'core-api-client/src/types/client/Recipient';

import { render } from '../../../../testUtils/render';
import { buildDefaultRecord } from '../../../../utils/buildDefaultRecord';
import { RecipientEditorComponent } from '../RecipientEditor.component';

jest.mock('../../../RecordEditor', () => {
  const { View, Text } = jest.requireActual('react-native');
  return {
    RecordEditor: ({
      form,
      record,
      onChange,
    }: {
      form: FormDefinition;
      record: Record;
      onChange: () => void;
    }) => (
      <View>
        <Text>{JSON.stringify(form)}</Text>
        <Text>{JSON.stringify(record)}</Text>
      </View>
    ),
  };
});

const makeFormWithRecord = (i: number): FormWithRecord<Recipient> => {
  const form = {
    id: `form-id-${i}`,
    code: 'form-code',
    databaseId: 'database-id',
    folderId: 'folder-id',
    name: `form-name-${i}`,
    formType: FormType.DefaultFormType,
    fields: [
      {
        id: `field-id-${i}`,
        name: `field-name-${i}`,
        code: '',
        description: '',
        required: false,
        key: false,
        fieldType: { text: {} },
      },
    ],
  };

  return {
    form,
    record: buildDefaultRecord(form),
  };
};

it('should match the snapshot', () => {
  const { toJSON } = render(
    <RecipientEditorComponent
      data={[
        makeFormWithRecord(1),
        makeFormWithRecord(2),
        makeFormWithRecord(3),
      ]}
    />,
  );
  expect(toJSON()).toMatchSnapshot();
});

it('should throw an error with less than 2 forms', () => {
  expect(() => {
    render(<RecipientEditorComponent data={[makeFormWithRecord(1)]} />);
  }).toThrow();
});
