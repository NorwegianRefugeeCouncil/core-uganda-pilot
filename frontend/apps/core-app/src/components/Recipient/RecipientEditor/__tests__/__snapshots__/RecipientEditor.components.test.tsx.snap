// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={Object {}}
  >
    <View
      dataSet={Object {}}
      style={Object {}}
    >
      <View
        _text={
          Object {
            "color": "lightText",
          }
        }
        accessible={true}
        collapsable={false}
        dataSet={Object {}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onHoverIn={[Function]}
        onHoverOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#24303E",
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          }
        }
      >
        <Text
          dataSet={Object {}}
          level="1"
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#fff",
              "fontFamily": "Roboto_400Regular",
              "fontSize": 18,
              "fontStyle": undefined,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 21,
              "textAlign": "auto",
              "textDecorationLine": undefined,
            }
          }
        >
          form-name-2
        </Text>
      </View>
      <View
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": "#F6F6F7",
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          }
        }
      >
        <View
          dataSet={Object {}}
          style={Object {}}
        >
          <View>
            <Text>
              {"id":"form-id-1","code":"form-code","databaseId":"database-id","folderId":"folder-id","name":"form-name-1","formType":"default","fields":[{"id":"field-id-1","name":"field-name-1","code":"","description":"","required":false,"key":false,"fieldType":{"text":{}}}]}
            </Text>
            <Text />
          </View>
          <View
            dataSet={Object {}}
            style={
              Object {
                "height": 16,
              }
            }
          />
          <View>
            <Text>
              {"id":"form-id-2","code":"form-code","databaseId":"database-id","folderId":"folder-id","name":"form-name-2","formType":"default","fields":[{"id":"field-id-2","name":"field-name-2","code":"","description":"","required":false,"key":false,"fieldType":{"text":{}}}]}
            </Text>
            <Text />
          </View>
        </View>
      </View>
    </View>
    <View
      dataSet={Object {}}
      style={
        Object {
          "height": 16,
        }
      }
    />
    <View
      dataSet={Object {}}
      style={Object {}}
    >
      <View
        _text={
          Object {
            "color": "lightText",
          }
        }
        accessible={true}
        collapsable={false}
        dataSet={Object {}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onHoverIn={[Function]}
        onHoverOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#24303E",
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          }
        }
      >
        <Text
          dataSet={Object {}}
          level="1"
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#fff",
              "fontFamily": "Roboto_400Regular",
              "fontSize": 18,
              "fontStyle": undefined,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 21,
              "textAlign": "auto",
              "textDecorationLine": undefined,
            }
          }
        >
          form-name-3
        </Text>
      </View>
      <View
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": "#F6F6F7",
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          }
        }
      >
        <View>
          <Text>
            {"id":"form-id-3","code":"form-code","databaseId":"database-id","folderId":"folder-id","name":"form-name-3","formType":"default","fields":[{"id":"field-id-3","name":"field-name-3","code":"","description":"","required":false,"key":false,"fieldType":{"text":{}}}]}
          </Text>
          <Text />
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
